# Task 29 — Average Download by Subscription Plan

## Business Question

How does average data usage differ between subscription plans?

## Analysis

Grouped customers by plan_name and calculated the average gb_downloaded for each subscription plan. The results were sorted in descending order by average data usage.

## Code

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
starlink_average_download_by_plan = starlink.groupby("plan_name").agg(
    avg_downloaded_gb = ("gb_downloaded","mean")
).sort_values(
    "avg_downloaded_gb",
    ascending = False
)
print(starlink_average_download_by_plan)

             avg_downloaded_gb
plan_name                     
Business           1546.626850
Roam               1531.217988
Residential        1507.878364
Mini               1503.507140


### Business Insight

Business customers have the highest average data usage at approximately 1,546.63 GB, followed by Roam (1,531.22 GB), Residential (1,507.88 GB), and Mini (1,503.51 GB). The differences between plans are relatively small, indicating that average data usage is fairly similar across subscription plans in this dataset.